### ISSUE TITLE
build_or_load(): TypeError: __int__ returned non-int (type NoneType)

### ISSUE THREAD

**Issue by chenqi1990** [calclavia/DeepJ#68]
`build_or_load()`
---------------------------------------------------------------------------
Traceback (most recent call last):
  File "util.py", line 36, in <module>
    build_or_load()
  File "util.py", line 15, in build_or_load
    models = build_models()
  File "/Users/chenqi/workspace/DeepJ/model.py", line 148, in build_models
    notes_out = naxis(time_out, chosen, style)
  File "/Users/chenqi/workspace/DeepJ/model.py", line 110, in f
    dense_layer_cache[l] = Dense(int(x.get_shape()[3]))
TypeError: __int__ returned non-int (type NoneType)

-----------------------------------------------------------------------------
```
shift_chosen = Lambda(lambda x: tf.pad(x[:, :, :-1, :], [[0, 0], [0, 0], [1, 0], [0, 0]]))(chosen)
# shift_chosen: shape=(?, 128, 48, 3) 
shift_chosen = Reshape((time_steps, NUM_NOTES, -1))(shift_chosen)
# shift_chosen: shape=(?, 128, 48, ?)
x = Concatenate(axis=3)([x, shift_chosen])
# x: shape=(?, 128, 48, ?)
```
then `int(x.get_shape()[3])` throw TypeError: __int__ returned non-int (type NoneType)

**Comment by chenqi1990** [comment]
@calclavia  Could you help me solve this problem?
And I can not understand the mean of [note_axis module](https://github.com/calclavia/DeepJ/blob/icsc/model.py#L91).

**Comment by slave2sync** [comment]
I also reproduced this with Keras 2.1.5. It seems that Keras 2.0 is compatible, but there are other problems there (see my issue https://github.com/calclavia/DeepJ/issues/69)

**Comment by chenqi1990** [comment]
@slave2sync Thank you so much! The bug is fixed with Keras 2.0 and Tensorflow 1.6.0.
